### ISSUE TITLE
Request for code review: option to render the current item as a link

### ISSUE THREAD

**Issue by nbudin** [jamesarosen/xebec#6]
I've added an option to make the current nav bar item a link (rather than a span as usual).  This is useful, for example, in sites where you want to show the same current tab for everything in a particular "section" of the site, but you'd like to be able to let users click the tab to get back to the front page of the section easily.

The way you use it is pretty simple:

``` ruby
nav_bar :main_menu do |nb|
  nb.nav_item :posts
  nb.nav_item :users
  nb.nav_item :comments

  nb.render_current_as_link = true
end
```

Patch and tests are on the render_current_as_link branch.  Let me know if you like it; if so, we can merge it to master and release a new version.

**Comment by jamesarosen** [comment]
I think the use case is totally valid, but (a) I'd like to avoid more configuration if possible and (b) if we go that route, it seems like it should be global configuration in a `config/initializers/xebec.rb` file.

**Comment by nbudin** [comment]
I definitely agree with avoiding more config, but I'm not sure how we'd do that without changing the behavior for existing apps (or is that even a problem?).  Global config isn't quite as flexible in that you can't set this behavior only for one particular nav bar, but that's probably an extreme edge case, and one I'm perfectly willing not to support.

Should I take a stab at making it global?

**Comment by jamesarosen** [comment]
I know Xebec supports different nav-bar renderers, though I've forgotten the exact syntax. What about making it an option on the renderer? For example, in `config/initializers/xebec.rb`:

``` ruby
Xebec.renderer = Xebec::NavBarRenderer.new({ current_is_link: true })
```

That way, if a user _really_ wants to do something different, she just has to use a different renderer.

**Comment by nbudin** [comment]
Unfortunately, I don't think that will work.  Xebec creates a new NavBarRenderer object for each bar you try to render, and it does it based on the `Xebec.renderer_class` config option.  So you can set the renderer class, but not options for the renderer.

Two additional, but sort of bad, options occur to me this:
1. We could add a new NavBarRenderer subclass that renders the current item as a link, and then you would set that as the value of `Xebec.renderer_class`.  Unfortunately, the design of NavBarRenderer doesn't make it terribly easy to factor this out without duplicating a bunch of the existing method, and I think it's sort of an orthogonal concern anyway: what if I want to use the WebAppThemeRenderer, for example, but still have the current item be a link?
2. We could change the API of NavBarRenderer to allow stuff like this.  One way to do it would be to add a `Xebec.renderer_options` hash in which you could specify that option (and then change all existing NavBarRenderer classes to support it).  But this isn't great either, because it probably requires changes to all existing NavBarRenderer subclasses.

On further thought, what about a compromise?  We could create a `Xebec.current_is_link=` method to set it up globally, and allow individual `nav_bar` blocks to override it using the API I created if they want.  That would neatly allow us to avoid extra code in the `nav_bar` blocks, while still making it possible to override it on a per-navbar basis.

**Comment by nbudin** [comment]
I've updated the branch to try out the compromise solution I alluded to above.  The `Xebec` module now has a `current_is_link` attribute, which is the global default, but can be overridden by individual navbars,

**Comment by jamesarosen** [comment]
That seems like the cleanest we're gonna get it. :+1:

**Comment by nbudin** [comment]
OK, I've merged it to master.  Would you mind doing a gem release with this?  (Or if you want to give me access to the rubygems project, I'm happy to do that too.)

**Comment by nbudin** [comment]
Released version 2.7.0 with this feature.
